Newsletters: add 286 (2024-01-24)

harding opened the pull request:
- Lede, releases/RCs, topic entries @harding 
- PR summaries @harding 
- Client/Services @bitschmidty

vostrnad commented:
Should this perhaps inlcude the [btcd disclosure](https://delvingbitcoin.org/t/disclosure-btcd-consensus-bugs-due-to-usage-of-signed-transaction-version/455)? Since it's a consensus bug I think it should be included in the earliest possible newsletter.

harding commented:
> Should this perhaps inlcude the [btcd disclosure](https://delvingbitcoin.org/t/disclosure-btcd-consensus-bugs-due-to-usage-of-signed-transaction-version/455)? Since it's a consensus bug I think it should be included in the earliest possible newsletter.

Good idea, I added a section to the top of the newsletter (our usual policy is that significant security disclosure go first).

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+## News
+
+- **Disclosure of fixed consensus failure in btcd:** "Niklas Gögge
+  used Delving Bitcoin to [announce][gogge btcd] a consensus failure in
```
Not a fan of "used Delving Bitcoin to announce", how about:
```suggestion
  [announced][gogge btcd] on Delving Bitcoin a consensus failure in
```

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+  transactions created before the fork would not be made invalid, the
+  relative time locks only apply to transactions with version numbers 2
+  and higher, allowing transactions with the orginal default of version
+  1 to remain valid with any inputs.  However, in the original Bitcoin
+  protocol, version numbers are signed integers, meaning negative
```
I think it's incorrect to say that version numbers are signed integers in the Bitcoin protocol. They are first and foremost a fixed-length byte sequence, and the only time they impact consensus they're treated as unsigned integers, so it's more a matter of how they're handled in the code.
```suggestion
  1 to remain valid with any inputs.  However, both Bitcoin Core and
  btcd treat version numbers as signed integers, meaning negative
```

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+  1 to remain valid with any inputs.  However, in the original Bitcoin
+  protocol, version numbers are signed integers, meaning negative
+  versions are possible.  The _reference implementation_ section of
+  [BIP68][] (but not the descriptive text) notes that "version 2 and
```
```suggestion
  [BIP68][] (but not the specification section) notes that "version 2 and
```

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+  [BIP68][] (but not the descriptive text) notes that "version 2 and
+  higher" is meant to apply to the version number cast from a signed
+  integer into an unsigned integer, ensuring that the rules apply to any
+  transaction that isn't version 1.
```
```suggestion
  transaction that isn't version 0 or 1.
```

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+    require follow the BIP68 rules but btcd would not.  In that case,
+    one node could reject the transaction (and any block that contained
+    it) and the other node could accept it (and its block), leading to a
+    chainsplit.  An attacker could use this to trick the operator of a
```
Per the style guide:
```suggestion
    chain split.  An attacker could use this to trick the operator of a
```

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+    invalid bitcoins.
+
+    The bug was privately disclosed to the btcd maintainers who fixed it
+    in the recent v0.24.0 release.  Anyone using btcd for consenus
```
```suggestion
    in the recent v0.24.0 release.  Anyone using btcd for consensus
```

vostrnad commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,189 @@
+  to transaction input sequence numbers.  To ensure any presigned
+  transactions created before the fork would not be made invalid, the
+  relative time locks only apply to transactions with version numbers 2
+  and higher, allowing transactions with the orginal default of version
```
```suggestion
  and higher, allowing transactions with the original default of version
```

murchandamus reviewed (approved):
LGTM

bitschmidty commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,233 @@
+
+  - *Trimming redirect:* instead of spending the value of all [trimmed HTLCs][topic trimmed
+    htlc] to fees in the commitment transaction, the combined value is
+    added to the value of the anchor output, ensuring the extra fees are
```
Maybe specify that its an ephemeral anchor output? (if I have it right)

bitschmidty commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,233 @@
+
+    - [BIN24-2][] Heretical Deployments by Anthony Towns, describing the
+      use of [Bitcoin Inquisition][bitcoin inquisition repo] for
+      proposed soft forks and other changes on the default signet.  See
```
```suggestion
      proposed soft forks and other changes on the default [signet][topic signet].  See
```

LarryRuane commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,233 @@
+  software, version numbers are signed integers, meaning negative
+  versions are possible.  The _reference implementation_ section of
+  [BIP68][] notes that "version 2 and
+  higher" is meant to apply to the version number cast from a signed
+  integer into an unsigned integer, ensuring that the rules apply to any
```
The word "cast" here refers to a fairly low-level programming concept that (I think) only applies to languages like C, C++, and java. The term does express the desired concept perfectly, but maybe for the benefit of readers less familiar with programming:
```suggestion
  higher" is meant to apply to the version number cast from a signed
  integer into an unsigned integer (for example, -2, or `0xfffffffe`, is interpreted as a very large number), ensuring that the rules apply to any
```

harding replied:
Good idea on providing more ideas.  I didn't like inserting a parenthetical here (breaks up the sentence flow too much), so I used a link to Wikipedia instead.

LarryRuane commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,233 @@
+
+- **Liana v4.0 released:**
+  [Liana v4.0][] was [released][liana blog] and includes support for [RBF fee
+  bumping][topic rbf], transaction cancelling using RBF, automatic [coin
```
```suggestion
  bumping][topic rbf], transaction canceling using RBF, automatic [coin
```
(American English spelling; "cancelling" is British English, so I'm not sure)
So far in the newsletters, we have 4 occurrences of "canceling" and zero of "cancelling".

LarryRuane commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,233 @@
+    Gögge discovered that btcd did not implement this conversion from
+    signed to unsigned integers, so it was possible to construct a
+    transaction with a negative version number which Bitcoin Core would
+    require follow the BIP68 rules but btcd would not.  In that case,
```
```suggestion
    require to follow the BIP68 rules but btcd would not.  In that case,
```

LarryRuane commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,233 @@
+  integer into an unsigned integer, ensuring that the rules apply to any
+  transaction that isn't version 0 or 1.
+
+    Gögge discovered that btcd did not implement this conversion from
```
not sure but maybe here and elsewhere:
```suggestion
    Gögge discovered that `btcd` did not implement this conversion from
```

vostrnad replied:
btcd is the name of the software, so backticks definitely aren't appropriate.

harding commented:
Made edits for all feedback (thanks @vostrnad @jirijakes @murchandamus @bitschmidty @LarryRuane !), reviewed @bitschmidty's section (thanks!), added remaining content, and added topic entries.  Thanks everyone!

glozow commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+    sufficient reserve, the only time they need a separate UTXO for
+    security reasons is when they want to accept an incoming payment.
+
+  - *Circular dependencies:* v3 relay depends on [cluster mempool][topic
```
v3 doesn't depend on cluster mempool?

glozow replied:
To elaborate, there isn't a circular dependency problem. The "imbue v3 logic on commitment transactions" idea was proposed by @TheBlueMatt in the LN spec meeting because he thinks it might take longer for LN to switch commitment transaction formats than for Bitcoin Core to merge cluster mempool, and LN development shouldn't block Bitcoin Core development.

glozow commented on _posts/en/newsletters/2024-01-24-newsletter.md:
```diff
@@ -0,0 +1,240 @@
+    sufficient reserve, the only time they need a separate UTXO for
+    security reasons is when they want to accept an incoming payment.
+
+  - *Circular dependencies:* v3 relay depends on [cluster mempool][topic
+    cluster mempool], cluster mempool depends on CPFP carve out being
+    removed, but carve out can't be removed until LN users switch to v3
+    relay.  Gregory Sanders [proposes][sanders transition] an
+    intermediate stage with temporary special treatment of transactions
+    that look like current anchors-style LN commitment transactions,
+    allowing Bitcoin Core to deploy cluster mempool and v3.  When
+    they've been widely deployed and all LN implementations have
+    upgraded to using them, the temporary special rules can be dropped.
```
(Apologies for the late review) proposed new wording:

```suggestion
  - *Imbued v3 logic:* In response to concerns voiced in the LN spec meeting that it may
    take a long time for LN to design, implement, and deploy these changes,
    Gregory Sanders [proposed][sanders transition] an
    intermediate stage with temporary special treatment of transactions
    that look like current anchors-style LN commitment transactions,
    allowing Bitcoin Core to deploy cluster mempool without being blocked by LN development.  When
    they've been widely deployed and all LN implementations have
    upgraded to using v3, the temporary special rules can be dropped.
```

harding replied:
@glozow thanks for catching this error and providing the rewording!

(For anyone who's curious, I made a bad inference from this sentence by Greg: "We need to roll out **updates to mempool/relay** to make the transition to v3 style transactions safe, so LN implementations can adopt it" (emphasis added).  I inferred the "updates to mempool" meant cluster mempool.  I'm guessing now that what he really meant was more in the line of package relay / package rbf.)

glozow replied:
Understandable. It's my bad I should have reviewed earlier. Thanks @bitschmidty for being flexible and making last-minute changes!
